rocketpy/motors/tank.py

```diff
@@ -410,6 +409,73 @@ def inertia(self):
+            raise ValueError(
+                f"The tank '{self.name}' is overfilled. The fluid volume is "
+                + "greater than the total volume of the tank.\n\t\t"
+                + "Try increasing the tank height and check out the fluid density"
+                + " values.\n\t\t"
+                + f"The fluid volume is {np.max(self.fluid_volume.y_array):.3f} m³ at "
+                + f"{self.fluid_volume.x_array[np.argmax(self.fluid_volume.y_array)]:.3f} s."
+            )
+        elif (self.fluid_volume < -1e-6).any():
+            raise ValueError(
+                f"The tank '{self.name}' is underfilled. The fluid volume is "
+                + "negative.\n\t\t"
+                + "Try increasing input fluid quantities and check out the fluid density"
+                + " values.\n\t\t"
+                + f"The fluid volume is {np.min(self.fluid_volume.y_array):.3f} m³ at "
+                + f"{self.fluid_volume.x_array[np.argmin(self.fluid_volume.y_array)]:.3f} s."
+            )
+
+    def _check_height_bounds(self):
+        """Checks if the tank is overfilled or underfilled. Raises a ValueError
+        if the tank is overfilled or underfilled.
+        """
+        top_tolerance = self.geometry.top + 1e-4
+        bottom_tolerance = self.geometry.bottom - 1e-4
+
+        if (self.liquid_height > top_tolerance).any():
+            raise ValueError(
+                f"The tank '{self.name}' is overfilled. "
+                + f"The liquid height is above the tank top.\n\t\t"
+                + "Try increasing the tank height and check out the fluid density"
+                + " values.\n\t\t"
+                + f"The liquid height is {np.max(self.liquid_height.y_array):.3f} m above "
+                + f"the tank top at {self.liquid_height.x_array[np.argmax(self.liquid_height.y_array)]:.3f} s."
+            )
+        elif (self.liquid_height < bottom_tolerance).any():
+            raise ValueError(
+                f"The tank '{self.name}' is underfilled. "
+                + "The liquid height is below the tank bottom.\n\t\t"
+                + "Try increasing input fluid quantities and check out the fluid density"
+                + " values.\n\t\t"
+                + f"The liquid height is {np.min(self.liquid_height.y_array):.3f} m below "
+                + f"the tank bottom at {self.liquid_height.x_array[np.argmin(self.liquid_height.y_array)]:.3f} s."
+            )
+        elif (self.gas_height > top_tolerance).any():
+            raise ValueError(
+                f"The tank '{self.name}' is overfilled. "
+                + "The gas height is above the tank top.\n\t\t"
+                + "Try increasing the tank height and check out the fluid density"
+                + " values.\n\t\t"
+                + f"The gas height is {np.max(self.gas_height.y_array):.3f} m above "
+                + f"the tank top at {self.gas_height.x_array[np.argmax(self.gas_height.y_array)]:.3f} s."
```

Comment:
`gas_height` is usually a negative value right? Will this check work as intended?

Reply:
I do think this check works: both `geometry.top` and `gas_height` are calculated with respect to the `Tank` geometry reference, so they could be both negative or positive.

If the `gas_height` surpasses the top tolerance (e.g. -0.9 > -1), the check would point the overfill.